src/gui/editing.cpp

```diff
@@ -607,6 +591,796 @@ void FurnaceGUI::doPaste(PasteMode mode, int arg, bool readClipboard, String cli
+      {
+        return ((0x02) << 8) | val;
+      }
+
+      else if(val >= 0xe0 && val < 0xf0)
+      {
+        return ((0xF2) << 8) | (val & 0xf);
+      }
+
+      else
+      {
+        return ((0xF2) << 8) | ((val & 0xf) / 2);
+      }
+
+      break;
+    }
+
+    case 'F': //who the fuck invented this...
+    {
+      if(val < 0xe0)
+      {
+        return ((0x01) << 8) | val;
+      }
+
+      else if(val >= 0xe0 && val < 0xf0)
+      {
+        return ((0xF1) << 8) | (val & 0xf);
+      }
+
+      else
+      {
+        return ((0xF1) << 8) | ((val & 0xf) / 2);
+      }
+
+      break;
+    }
+
+    case 'G':
+    {
+      return ((0x03) << 8) | (val);
+      break;
+    }
+
+    case 'H':
+    {
+      return ((0x04) << 8) | (val);
+      break;
+    }
+
+    case 'J':
+    {
+      return ((0x00) << 8) | (val);
+      break;
+    }
+
+    case 'K':
+    {
+      return ((0x0a) << 8) | (val) | ((0x04) << 24); //Axy + 400
+      break;
+    }
+
+    case 'L':
+    {
+      return ((0x0a) << 8) | (val) | ((0x03) << 24); //Axy + 300
+      break;
+    }
+
+    case 'O':
+    {
+      return ((0x90) << 8) | (val);
+      break;
+    }
+
+    case 'Q':
+    {
+      return ((0xC0) << 8) | (val & 0xf);
+      break;
+    }
+
+    case 'R':
+    {
+      return ((0x07) << 8) | (val & 0xf);
+      break;
+    }
+
+    case 'S':
+    {
+      switch(val >> 4)
+      {
+        case 2:
+        {
+          return ((0xE5) << 8) | ((val & 0xf) << 4);
+          break;
+        }
+
+        case 8:
+        {
+          return ((0x80) << 8) | ((val & 0xf) << 4);
+          break;
+        }
+
+        case 0xC:
+        {
+          return ((0xFC) << 8) | (val & 0xf);
+          break;
+        }
+
+        case 0xD:
+        {
+          return ((0xFD) << 8) | (val & 0xf);
+          break;
+        }
+
+        default: break;
+      }
+
+      break;
+    }
+
+    case 'T':
+    {
+      return ((0xF0) << 8) | (val & 0xf);
+      break;
+    }
+
+    case 'U':
+    {
+      return ((0x04) << 8) | MAX(1, ((val & 0xf0) >> 4 / 4 << 4)) | MAX(1, ((val & 0xf) / 4));
+      break;
+    }
+
+    case 'X':
+    {
+      return ((0x80) << 8) | (val);
+      break;
+    }
+
+    default: return 0; break;
+  }
+
+  return 0;
+}
+
+uint64_t convert_effect_openmpt_xm(char symbol, uint16_t val)
+{
+  return convert_effect_openmpt_mod(symbol, val);
+  //other effects too obscure bruh
+}
+
+uint64_t convert_effect_openmpt_it(char symbol, uint16_t val)
+{
+  return convert_effect_openmpt_s3m(symbol, val);
+  //other effects too obscure bruh
+}
+
+uint64_t convert_effect_openmpt_mptm(char symbol, uint16_t val)
+{
+  if(symbol == ':')
+  {
+    return ((0xED) << 8) | ((val & 0xf0) >> 4) | ((0xEC) << 24) | ((((val & 0xf0) >> 4) + (val & 0xf)) << 16);
+  }
+
+  return convert_effect_openmpt_it(symbol, val);
+}
+
+void FurnaceGUI::doPasteOpenMPT(PasteMode mode, int arg, bool readClipboard, String clipb, std::vector<String> data, int openmpt_format)
```

Comment:
Rename to doPasteMPT or doPasteModPlug (see third clause of BSD license for more info on why)

Rename openmpt_format to mptFormat